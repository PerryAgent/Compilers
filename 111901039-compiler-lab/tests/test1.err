Pretty Printing IR:

.globl main
main:
li t1, 0
li t2, 100
Label_No_1:
bgt t1, t2, Label_No_2
move t3 t1
move t9 t3
li t8, 1
syscall
addi t1, t1, 1
j Label_No_1
Label_No_2:
li t8, 10
syscall


Basic Blocks:

.globl main

main:
li $t1, 0
li $t2, 100

Label_No_1:
bgt $t1, $t2, Label_No_2

move $t3 $t1
move $a0 $t3
li $v0, 1
syscall
addi $t1, $t1, 1
j Label_No_1


Label_No_2:
li $v0, 10
syscall

