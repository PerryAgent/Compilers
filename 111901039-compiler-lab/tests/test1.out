.globl main
main:
li $t1, 0
li $t2, 100
Label_No_1:
bgt $t1, $t2, Label_No_2
move $t3 $t1
move $a0 $t3
li $v0, 1
syscall
addi $t1, $t1, 1
j Label_No_1
Label_No_2:
li $v0, 10
syscall
